extensions/todoist/src/components/FilterTasks.tsx

```diff
@@ -0,0 +1,75 @@
+import { List, ActionPanel, Action, Icon } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import { useEffect, useState } from "react";
+
+import { Task, getFilterTasks } from "../../src/api";
+import { filterSort } from "../helpers/filters";
+import { QuickLinkView, ViewMode } from "../home";
+import useCachedData from "../hooks/useCachedData";
+import useViewTasks from "../hooks/useViewTasks";
+
+import CreateViewAction from "./CreateViewAction";
+import TaskListSections from "./TaskListSections";
+
+type FilterTasksProps = { name: string; quickLinkView?: QuickLinkView };
+
+function FilterTasks({ name, quickLinkView }: FilterTasksProps) {
+  const [cachedData] = useCachedData();
+  const filters = cachedData?.filters;
+  const filter = filters?.find((filter: { name: string }) => filter.name === name);
+  const query = filter?.query || "";
+  const [tasks, setTasks] = useState<Task[]>([]);
+
+  const getFilterTasksCached = async (query: string) => {
+    const filterTasks = await getFilterTasks(query);
+    return filterTasks;
+  };
+
+  const { isLoading, data, revalidate } = useCachedPromise(getFilterTasksCached, [query]);
+
+  useEffect(() => {
+    if (query) {
+      revalidate();
+    }
+  }, [query]);
```

Comment:
This doesn't seem to be needed as you've already past `query` as a dependency in `useCachedPromise`.

Reply:
@thomaslombart  thanks for pointing that out .